src/main/java/frc/robot/subsystems/Arm/ArmSubsystem.java

```diff
@@ -0,0 +1,340 @@
+        }
+      } else {
+        this.stow();
+        if (piece == gamePiece.CUBE) {
+          finalState = ArmState.LOW_CUBE;
+        } else {
+          finalState = ArmState.LOW_CONE;
+        }
+        useFinal = true;
+      }
+    } else {logger.info("cannot place nonexistent game piece");}
+  }
+
+  public void shelf(gamePiece piece) {
+    if (piece != gamePiece.NONE) {
+      if (currState == ArmState.STOW) {
+        currState = ArmState.PARALLEL_TRANS;
+        if (piece == gamePiece.CUBE) {
+          desiredState = ArmState.SHELF_CUBE;
+          shoulderSubsystem.setPos(ArmConstants.kCubeShelfShoulderPos);
+          extendoSubsystem.setPos(ArmConstants.kCubeShelfExtendoPos);
+          wristSubsystem.setPos(ArmConstants.kCubeShelfWristPos);
+        } else {
+          desiredState = ArmState.SHELF_CONE;
+          shoulderSubsystem.setPos(ArmConstants.kConeShelfShoulderPos);
+          extendoSubsystem.setPos(ArmConstants.kConeShelfExtendoPos);
+          wristSubsystem.setPos(ArmConstants.kConeShelfWristPos);
+        }
+      } else {
+        this.stow();
+        if (piece == gamePiece.CUBE) {
+          finalState = ArmState.SHELF_CUBE;
+        } else {
+          finalState = ArmState.SHELF_CONE;
+        }
+        useFinal = true;
+      }
+    } else {logger.info("cannot pickup nonexistent game piece");}
+  }
+
+  public void floor(gamePiece piece, boolean isUpright) {
+    // if it's a cube, set isUpright to whatever! freedom!!!
+    if (piece != gamePiece.NONE) {
+      if (currState == ArmState.STOW) {
+        currState = ArmState.PARALLEL_TRANS;
+        if (piece == gamePiece.CUBE) {
+          desiredState = ArmState.FLOOR_CUBE;
+          shoulderSubsystem.setPos(ArmConstants.kCubeFloorShoulderPos);
+          extendoSubsystem.setPos(ArmConstants.kCubeFloorExtendoPos);
+          wristSubsystem.setPos(ArmConstants.kCubeFloorWristPos);
+        } else {
+          desiredState = ArmState.FLOOR_CONE;
+          if (isUpright) {
+            shoulderSubsystem.setPos(ArmConstants.kConeFloorUprightShoulderPos);
+            extendoSubsystem.setPos(ArmConstants.kConeFloorUprightExtendoPos);
+            wristSubsystem.setPos(ArmConstants.kConeFloorUprightWristPos);
+          } else {
+            shoulderSubsystem.setPos(ArmConstants.kConeFloorShoulderPos);
+            extendoSubsystem.setPos(ArmConstants.kConeFloorExtendoPos);
+            wristSubsystem.setPos(ArmConstants.kConeFloorWristPos);
+          }
+        }
+      } else {
+        this.stow();
+        if (piece == gamePiece.CUBE) {
+          finalState = ArmState.FLOOR_CUBE;
+        } else {
+          if (isUpright) {
+            finalState = ArmState.FLOOR_CONE_UPRIGHT;
+          } else {
+            finalState = ArmState.FLOOR_CONE;
+          }
+        }
+        useFinal = true;
+      }
+    } else {logger.info("cannot pickup nonexistent game piece");}
+  }
+
+  public void yoshi(gamePiece gamePiece) {
```

Comment:
need STOW check, also consider a duplicate setpoint to wrist/shoulder STOW positions inside of this method (if YOSHI does in fact use shoulder/wrist positions for stow